File: best_time_stock/best_time_stock.py
def best_time_stock(prices):
    if(len(prices) == 0):
        return 0
    min_price = prices[0]
    max_profit = 0
    for i in range(1, len(prices)):
        if(prices[i] < min_price):
            min_price = prices[i]
        else:
            profit = prices[i] - min_price
            if(profit > max_profit):
                max_profit = profit
    return max_profit

prices = [7,1,5,3,6,4]
print("The Max Profit is :", best_time_stock(prices))
            

File: best_time_stock/best_time_stock.ts
export {};

function bestTimeStock(prices: number[]): number {
  if (prices.length == 0) return 0;
  let min_price = prices[0];
  let max_profit = 0;
  for (let i = 1; i < prices.length; i++) {
    if (prices[i] < min_price) min_price = prices[i];
    else {
      const profit = prices[i] - min_price;
      if (profit > max_profit) max_profit = profit;
    }
  }
  return max_profit;
}

const prices = [7, 1, 5, 3, 6, 4];
console.log("The Max price is : ", bestTimeStock(prices));
